### PR TITLE
Remove OJDBC and resin artifacts from ci-dependencies driectory as they as not Apache license compatible

### PR DESCRIPTION
**Pull request by michaelsembwever**
With this approach the user can onetime manually download them, and the maven reactor will automatically build the corresponding plugins.

OJDBC is OTN which requires a user click-through agreement.
Resin is GPLv3, meaning any derivative work must also be GPLv3 licensed.
https://www.apache.org/licenses/GPL-compatibility.html

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -318,4 +278,89 @@
...
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+        <profile>
+            <!-- resin is not Apache license compatible, so it must be manually downloaded first -->
+            <id>install-resin</id>
```
Comment (ascrutae) [inline pom.xml]: I think the `resin-3` plugin should be removed too.

File: apm-sniffer/apm-sdk-plugin/pom.xml [hunk 2]
```diff
@@ -202,4 +200,31 @@
...
+                    <exists>${basedir}/../../ci-dependencies/ojdbc14-10.2.0.4.0.jar</exists>
+                </file>
+            </activation>
+            <modules>
+                <module>oracle-10.x-plugin</module>
+            </modules>
+        </profile>
+        <profile>
+            <!-- resin is not Apache license compatible, so it must be manually downloaded first -->
+            <id>include-resin</id>
```
Comment (ascrutae) [inline apm-sniffer/apm-sdk-plugin/pom.xml]: I think the `resin-3` plugin should be removed too

Comment (michaelsembwever) [reply apm-sniffer/apm-sdk-plugin/pom.xml]: done.

**Comment by michaelsembwever** [comment]
> BTW we need to adjust /docs/Supported-list.md document:
>
> Add Optional at Resin and Oracle driver plugins. Notice people these plugins are not in official releases, based on ASF license policy.
> Link to Build Guide. Help people to build the plugin by theirself.

done. let me know when i'm free to merge.

**Comment by michaelsembwever** [comment]
rebased off master.

**Comment by johnament** [comment]
Note that OJDBC drivers are now available via maven via the Oracle Maven Repository - https://blogs.oracle.com/dev2dev/get-oracle-jdbc-drivers-and-ucp-from-oracle-maven-repository-without-ides .  So this means you don't need to have the extra install step, but it is a good idea to put it in a dedicated profile to hide it away.

**Pull request merged**